front/src/components/PageNavigation.js

```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { setPageQuery } from "../actions";
+
+export default function PageNavigation() {
+  const numOfProducts = useSelector((state) => state.productCount);
+  const numOfPages = Math.ceil(numOfProducts / 9);
+  const pageLinks = [];
+  const currentPage = useSelector((state) => state.currentPage);
+
+  const dispatch = useDispatch();
+
+  const handleClick = (page) => {
+    dispatch(setPageQuery(page));
+  };
+
+  for (let i = 1; i <= numOfPages; i++) {
```

Comment:
Could you use `map` here or some other abstracted function to reduce these side-effects?